src/applications/facility-locator/actions/index.js

```diff
@@ -1,148 +1,236 @@
   };
-}
-
-export function searchWithBounds(bounds, facilityType, serviceType, page = 1) {
-  const params = compact([
-    ...bounds.map(c => `bbox[]=${c}`),
-    facilityType ? `type=${facilityType}` : null,
-    ['health', 'benefits'].includes(facilityType) && serviceType
-      ? `services[]=${serviceType}`
-      : null,
-    `page=${page}`,
-  ]).join('&');
-  const url = `${api.url}?${params}`;
-
-  return dispatch => {
+};
+
+/**
+ * Gets the details of a single Community Care Provider
+ * 
+ * @param {string} id The NPI/Tax ID of a specific provider
+ */
+export const fetchProviderDetail = (id) => {
+  return (dispatch) => {
     dispatch({
-      type: 'SEARCH_STARTED',
+      type: SEARCH_STARTED,
       payload: {
-        page,
-        searchBoundsInProgress: true,
-      },
+        active: true,
+      }
     });
 
-    return fetch(url, api.settings)
-      .then(res => res.json())
-      .then(
-        data => {
-          dispatch({ type: 'FETCH_VA_FACILITIES', payload: data });
-        },
-        err => dispatch({ type: 'SEARCH_FAILED', err }),
-      );
+    return LocatorApi.fetchProviderDetail(id)
+      .then(data => dispatch({ type: FETCH_LOCATION_DETAIL, payload: data }))
+      .catch(error => dispatch({ type: SEARCH_FAILED, error }));
   };
-}
+};
+
+/**
+ * Find which locations exist within the given bounding box's area.
+ * 
+ * Allows for filtering on location types and services provided.
+ * 
+ * @param {{bounds: number[], facilityType: string, serviceType: string, page: number}} 
+ */
+export const searchWithBounds = ({ bounds, facilityType, serviceType, page = 1 }) => {
+  const needsAddress = [LocationType.CC_PROVIDER, Location.ALL];
+  return (dispatch) => {
+    if (needsAddress.includes(facilityType) && ccLocatorEnabled()) { // Remove Feature-flag when going live.
+      reverseGeocodeBox(bounds).then(address => {
+        if (!address) {
+          dispatch({ type: SEARCH_FAILED, error: 'Reverse geocoding failed. See previous errors or network log.' });
+          return null;
+        }
+
+        return fetchLocations(address, bounds, facilityType, serviceType, page, dispatch);
+      });
+    } else {
+      return fetchLocations(null, bounds, facilityType, serviceType, page, dispatch);
+    }
+
+    return null;
+  };
+};
+
+/**
+ * Handles the actual API call to get the type of locations closest to `address`
+ * and/or within the given `bounds`.
+ * 
+ * @param {string=} address Address of the center-point of the search area
+ * @param {number[]} bounds Geo-coords of the bounding box of the search area
+ * @param {string} locationType (see config.js for valid types)
+ * @param {string} serviceType (see config.js for valid types)
+ * @param {number} page What page of results to request
+ * @param {Function} dispatch Redux's dispatch method
+ */
+const fetchLocations = (address = null, bounds, locationType, serviceType, page, dispatch) => {
```

Comment:
Yea, I remember us talking about this on the previous PR. However, I am a little confused. From how I read it you are suggesting we take advantage of `redux-thunk` so we can pass a function instead of an action-object.

But looking at how `redux-thunk` [handles](https://github.com/reduxjs/redux-thunk/blob/master/src/index.js) getting a function:
```js
function createThunkMiddleware(extraArgument) {
  return ({ dispatch, getState }) => next => action => {
    if (typeof action === 'function') {
      return action(dispatch, getState, extraArgument);
    }

    return next(action);
  };
}
```
It's simply passing `dispatch` along itself. I remember agreeing it seemed odd to pass the `dispatch` function along as an argument, but if that's all the Thunk middleware is doing is it really so odd?